File: Frontend/src/graphql/graphql.ts
import { GraphQLClient } from 'graphql-request'
import gql from "graphql-tag"

const url = 'http://localhost:8080/graphql'

export const graphqlClientAuth = (token: string) => {
	return new GraphQLClient(url, {
		headers: {
			authorization: `Bearer ${token}`,
		}
	})
}

export const graphqlClient = new GraphQLClient(url)

export const GET_PALLETS = gql`
  query GetAllPallets {
    getAllPallets {
        serialNumber
        weight
        product {
          title
          weight
        }
    }
}
`

export const UPDATE_PRODUCT = gql`
  mutation Mutation($serialNumber: String!, $product: ProductInput!) {
  updateProductOfPallet(serialNumber: $serialNumber, product: $product) {
    serialNumber
  }
}
`

export const LOGIN = gql`
  mutation Mutation($email: String!, $password: String!) {
  login(email: $email, password: $password) {
    authToken {
      token
      user
    }
    isAuthenticated
  }
}
`

File: Edge/graphql.py
import os
import json
import requests
from dotenv import load_dotenv

GET_PALLETS = """query Query {
  getAllPallets {
    serialNumber
    weight
  }
}"""

CREATE_PALLET = """mutation CreatePallet($pallet: PalletInput!) {
  createPallet(pallet: $pallet) {
    serialNumber,
    weight
  }
}"""

DELETE_PALLET = """mutation Mutation($serialNumber: String!) {
  deletePallet(serialNumber: $serialNumber) {
    serialNumber
  }
}"""

UPDATE_PALLET_WEIGHT = """mutation Mutation($serialNumber: String!, $weight: Int!) {
  updatePalletWeight(serialNumber: $serialNumber, weight: $weight) {
    serialNumber,
    weight
  }
}"""


load_dotenv()
token = os.environ.get("TOKEN")
url = os.environ.get("URL")
headers = {"Authorization": f'Bearer {token}'}


def graphql_request(query: str, variables: json):
    return requests.post(
        url,
        json={
            'query': query,
            'variables': variables
        },
        headers=headers,
    )
